dipy/reconst/ivim.py

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/python
+""" Classes and functions for fitting ivim model """
+from __future__ import division, print_function, absolute_import
+import warnings
+import functools
+import numpy as np
+from scipy.optimize import curve_fit
+
+from dipy.core.gradients import gradient_table
+from dipy.reconst.base import ReconstModel
+from dipy.reconst.dti import _min_positive_signal
+
+
+def ivim_function(bvals, S0, f, D_star, D):
+    S = np.vectorize(lambda b, S0, f, D_star, D: S0 *
```

Comment:
Just curious from an implementation standpoint why you are using vectorize and lambda rather than straight up evaluation here.